tests/src/python/test_qgsproviderconnection_postgres.py

```diff
@@ -963,6 +962,38 @@ def test_rename_field(self):
         fields = conn.fields("qgis_test", "rename_field")
         self.assertEqual(fields.names(), ["id", "geom", "new_column"])
 
+    def test_move_table_to_schema(self):
+        """Test that table can be moved to another schema."""
+
+        md = QgsProviderRegistry.instance().providerMetadata("postgres")
+        conn = md.createConnection(self.uri, {})
+
+        sql = """
+        DROP TABLE IF EXISTS qgis_test.table_to_move;
+        CREATE TABLE qgis_test.table_to_move (
+            id SERIAL PRIMARY KEY,
+            geom geometry(Geometry,4326)
+        );
+        CREATE SCHEMA schema_test;
+        """
+
+        conn.executeSql(sql)
+
+        # test table exist
+        table = conn.table("qgis_test", "table_to_move")
+        self.assertEqual(table.tableName(), "table_to_move")
+
+        # move table to another schema
+        conn.moveTableToSchema(
+            "qgis_test",
+            "table_to_move",
+            "schema_test",
+        )
+
+        # test moved table exist in the schema
+        table = conn.table("schema_test", "table_to_move")
+        self.assertEqual(table.tableName(), "table_to_move")
+
 
```

Comment:
Can you also test what happens when an error occurs, eg the table doesn't exist? It should raise an exception in those situations.